src/zone.ts

```diff
@@ -179,7 +196,7 @@ export interface ZoneExportCallback {
  *
  * const zone = dns.zone('zone-id');
  */
-class Zone extends ServiceObject<Zone> {
+class Zone extends ZoneServiceObject {
```

Comment:
Any concerns with this messing up the prototype chain?  Making sure this ain't a breaking change.

Reply:
Technically the prototype chain isn't changing, `ZoneServiceObject` is just a workaround to allow us to override the `create` signature.

Reply:
I think my only concern would be, will ilt still be an `instanceof ServiceObject`, for someone consuming the library.

Let's just confirm this, and if so I don't think this is breaking.

Reply:
@bcoe we actually have a [unit test](https://github.com/googleapis/nodejs-dns/blob/master/test/zone.ts#L165) that specifically tests for this. `ZoneServiceObject` is literally just `ServiceObject`, but giving it an alias allows me to cast it as something else.